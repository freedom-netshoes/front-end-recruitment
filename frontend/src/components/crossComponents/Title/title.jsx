import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import CSSModules from 'react-css-modules'

import styles from './title.less'

@CSSModules(styles)
export default class Title extends Component {
  static defaultProps = {
    title: ''
  };

  static propTypes = {
    title: PropTypes.string
  };

  render () {
    return (
      <h3 className='title'> {this.props.title}</h3>
    )
  }
}
