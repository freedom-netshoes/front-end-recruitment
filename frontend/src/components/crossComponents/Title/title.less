@import '../../../styles/_less-variables.less';
@import '../../../styles/_less-mixins.less';
:global {
    #products {
        .title:hover {
            .fontSemiBold();
        }
        .title {
            color: @gray;
            font-size: 14px;
            margin-top: 8px;
        }
    }
}

:global {
    #cart-shop {
        .title {
            .fontSemiBold();
            font-size: 14px;
            color: @white; // margin-bottom: 8px;
        }
    }
}