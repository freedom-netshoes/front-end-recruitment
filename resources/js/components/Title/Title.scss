@import '../../../scss/shared';

:global(#products) {

  :global(li a:hover) .title {
    text-decoration: underline;
  }

  .title {
    color     : $gray;
    font-size : 14px;
    margin-top: 8px;
  }
}

:global(#cart) {

  :global(.delete) .title {
    @include opacity(0.2);
    text-decoration: line-through;
  }

  .title {
    @include open-sans-semi-bold();
    @include transition();
    color: white;
    font-size  : 14px;
    line-height: 18px;
  }
}