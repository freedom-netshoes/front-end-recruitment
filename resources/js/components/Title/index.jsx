import React, {Component, PropTypes} from 'react';
import CSSModules from 'react-css-modules';
import styles from './Title.scss';

@CSSModules(styles)
class Title extends Component {

  static defaultProps = {
    children: ''
  };

  static propTypes = {
    children: PropTypes.node
  };

  render () {
    return (
      <h4 styleName='title'>
        {this.props.children}
      </h4>
    );
  }
}

export default Title;
