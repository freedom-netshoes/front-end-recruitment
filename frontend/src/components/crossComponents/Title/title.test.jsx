/* global describe, it, expect */
import React from 'react'
import Title from './title'
import { shallow } from 'enzyme'

describe('Testing <Title /> component', () => {
  it('renders the Title component', () => {
    const component = shallow(<Title />)
    expect(component).toHaveLength(1)
  })

  it('Testing <Title /> snapshop', () => {
    const component = shallow(<Title {...{
      title: 'Camisa Nike Corinthians II'
    }} />)
    expect(component).toMatchSnapshot()
  })
})
