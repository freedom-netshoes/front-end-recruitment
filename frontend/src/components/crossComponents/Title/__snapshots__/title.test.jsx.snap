// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Title /> component Testing <Title /> snapshop 1`] = `
<h3
  className="title"
>
   
  Camisa Nike Corinthians II
</h3>
`;
